jctools-core/src/main/java/org/jctools/queues/atomic/BaseLinkedAtomicQueue.java

```diff
@@ -53,6 +53,57 @@ protected final void spConsumerNode(LinkedQueueAtomicNode<E> node) {
         throw new UnsupportedOperationException();
     }
 
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This method is only safe to call from the consumer thread!
+     */
+    @Override
+    public final boolean remove(Object o) {
+        if (null == o) {
+            return false; // Null elements are not permitted, so null will never be removed.
+        }
+
+        for (LinkedQueueAtomicNode<E> prevConsumerNode = null, currConsumerNode = getNextConsumerNode(lpConsumerNode());
+             currConsumerNode != null;
+             prevConsumerNode = currConsumerNode, currConsumerNode = getNextConsumerNode(currConsumerNode)) {
+            if (o.equals(currConsumerNode.lpValue())) {
+                LinkedQueueAtomicNode<E> nextNode = getNextConsumerNode(currConsumerNode);
+                if (nextNode != null && prevConsumerNode != null) {
+                    // We are removing an interior node.
+                    prevConsumerNode.soNext(nextNode);
+
+                    // Avoid GC nepotism because we are discarding the current node.
+                    currConsumerNode.spValue(null);
+                    currConsumerNode.soNext(null);
+                } else if (prevConsumerNode == null) {
+                    // We are removing the consumer node.
+                    currConsumerNode.spValue(null);
+                    spConsumerNode(currConsumerNode);
```

Comment:
> So same nepotism issue .... I think the prevConsumerNode should initially be consumerNode, and not null;

Good catch ... the "token" consumer node's next pointer was not being set to null. I'm not sure if the code can be the same as the case before, because the producer may have touched the `currConsumerNode.next`.